app/views/nic/_base_form.html.erb

```diff
@@ -17,16 +17,18 @@
 <%= text_f f, :name, :label => _("DNS name"), :value => f.object.shortname,
               :help_inline => _("Primary interface's DNS name and domain define host's FQDN"),
               :class => :interface_name %>
-<%= select_f f, :domain_id, accessible_domains, :id, :to_label,
-             { :include_blank => accessible_domains.any? ? true : _("No domains")},
-             { :disabled => accessible_domains.empty? ? true : false,
-               :help_inline => :indicator,
-               :class => :interface_domain, :'data-url' => domain_selected_hosts_path } %>
-<%= select_f f, :subnet_id, accessible_subnets, :id, :to_label,
-             { :include_blank => accessible_subnets.any? ? true : _("No subnets")},
-             { :disabled => accessible_subnets.empty? ? true : false,
-               :help_inline => :indicator,
-               :class => :interface_subnet, :'data-url' => freeip_subnets_path } %>
+<% if SETTINGS[:unattended] %>
+  <%= select_f f, :domain_id, accessible_domains, :id, :to_label,
```

Comment:
If user can't create domain it should not exist in DB and hostname is saved as FQDN. This can happen if user used Foreman in unattended and then switch to !unattended. Also displaying select box with only "No domains" option in 99% might be confusing. Cleaning mess is a good point but I'd rather keep users making to switch to unattended mode, clean mess and switch back.